### PR TITLE
[DOCS] Adds compatibility matrix to docs and readme

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds a compatibility matrix to the Overview section of the Python book and the repo readme. It also fixes the markup of the links at the top of the Installation page.

### Preview

* [Compatibility](https://elasticsearch-py_bk_2458.docs-preview.app.elstc.co/guide/en/elasticsearch/client/python-api/master/overview.html#_compatibility)
* [Installation](https://elasticsearch-py_bk_2458.docs-preview.app.elstc.co/guide/en/elasticsearch/client/python-api/master/installation.html)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -75,9 +75,22 @@ Usage
...
++-----------------------+-------------------------+-----------+
+| Elasticsearch Version | Elasticsearch-py Branch | Supported |
++=======================+=========================+===========+
+| main                  | main                    |           |
+| 8.x                   | 8.x                     | 8.x       |
+| 7.x                   | 7.x                     | 7.17      |
++-----------------------+-------------------------+-----------+
 
 If you have a need to have multiple versions installed at the same time older
 versions are also released as ``elasticsearch2`` and ``elasticsearch5``.
```
Comment (pquentin) [inline README.rst:96]:
```suggestion
If you have a need to have multiple versions installed at the same time older
versions are also released as ``elasticsearch7`` and ``elasticsearch8``.
```

File: docs/guide/overview.asciidoc [hunk 2]
```diff
@@ -10,9 +10,22 @@ on https://elasticsearch-py.readthedocs.io[Read the Docs].
...
+|===
+| Elasticsearch Version | Elasticsearch-py Branch   | Supported
+
+| main                  | main                      | 
+| 8.x                   | 8.x                       | 8.x
+| 7.x                   | 7.x                       | 7.17
+|===
 
 If you have a need to have multiple versions installed at the same time older
 versions are also released as `elasticsearch2` and `elasticsearch5`.
```
Comment (pquentin) [inline docs/guide/overview.asciidoc:31]:
```suggestion
If you have a need to have multiple versions installed at the same time older
versions are also released as `elasticsearch7` and `elasticsearch8`.
```

**Review by pquentin — COMMENTED** [review]
Thanks! Should we also update https://github.com/elastic/elasticsearch-py/blob/main/docs/sphinx/index.rst#compatibility?

**Comment by szabosteve** [comment]
> Should we also update https://github.com/elastic/elasticsearch-py/blob/main/docs/sphinx/index.rst#compatibility?

Yes, definitely! Thank you for the heads-up! I updated via [625c6fb](https://github.com/elastic/elasticsearch-py/pull/2458/commits/625c6fb3c0f676241d27f4f84b766fa2d152b127).

**Review by pquentin — COMMENTED** [review]
The tables do not render correctly in reST:

 * Read the Docs: https://elasticsearch-py--2458.org.readthedocs.build/en/2458/#compatibility
 * GitHub: https://github.com/elastic/elasticsearch-py/blob/625c6fb3c0f676241d27f4f84b766fa2d152b127/README.rst

According to https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#tables, you're missing the separation between the lines (`+---+---+---+`).

File: README.rst [hunk 3]
```diff
@@ -75,12 +75,25 @@ Usage
...
+communicating with greater or equal minor versions of Elasticsearch without
+breaking. It does not mean that the clients automatically support new features
+of newer Elasticsearch versions; it is only possible after a release of a new
+client version. For example, a 8.12 client version won't automatically support
+the new features of the 8.13 version of Elasticsearch, the 8.13 client version
+is required for that. Elasticsearch language clients are only backwards
+compatible with default distributions and without guarantees made.
+
++-----------------------+-------------------------+-----------+
+| Elasticsearch Version | Elasticsearch-py Branch | Supported |
```
Comment (pquentin) [inline README.rst:88]:
```suggestion
| Elasticsearch version | elasticsearch-py branch | Supported |
```

File: docs/guide/overview.asciidoc [hunk 4]
```diff
@@ -10,12 +10,25 @@ on https://elasticsearch-py.readthedocs.io[Read the Docs].
...
+communicating with greater or equal minor versions of {es} without breaking. It
+does not mean that the clients automatically support new features of newer
+{es} versions; it is only possible after a release of a new client version. For
+example, a 8.12 client version won't automatically support the new features of
+the 8.13 version of {es}, the 8.13 client version is required for that. {es}
+language clients are only backwards compatible with default distributions and
+without guarantees made.
+
+|===
+| Elasticsearch Version | Elasticsearch-py Branch   | Supported
```
Comment (pquentin) [inline docs/guide/overview.asciidoc:23]:
```suggestion
| Elasticsearch version | elasticsearch-py branch   | Supported
```

File: docs/sphinx/index.rst [hunk 5]
```diff
@@ -34,9 +34,22 @@ Read more about `how to use asyncio with this project <async.html>`_.
...
+communicating with greater or equal minor versions of Elasticsearch without
+breaking. It does not mean that the clients automatically support new features
+of newer Elasticsearch versions; it is only possible after a release of a new
+client version. For example, a 8.12 client version won't automatically support
+the new features of the 8.13 version of Elasticsearch, the 8.13 client version
+is required for that. Elasticsearch language clients are only backwards
+compatible with default distributions and without guarantees made.
+
++-----------------------+-------------------------+-----------+
+| Elasticsearch Version | Elasticsearch-py Branch | Supported |
```
Comment (pquentin) [inline docs/sphinx/index.rst:47]:
```suggestion
| Elasticsearch version | elasticsearch-py branch | Supported |
```

**Review by pquentin — COMMENTED** [review]
Also, can you please change the casing as below? `elasticsearch-py` isn't capitalized. Thanks and sorry for all the nitpicks.

**Comment by szabosteve** [comment]
Thanks for all the work on this, @pquentin! BTW, tables are killing me.

**Pull request merged**